plugins/lookup/passwordstore.py

```diff
@@ -343,19 +387,41 @@ def run(self, terms, variables, **kwargs):
             'length': 16,
             'backup': False,
             'missing': 'error',
+            'lock': self.get_option('lock'),
+            'locktimeout': self.get_option('locktimeout'),
         }
 
+        tmpdir = os.environ.get('TMPDIR', '/tmp')
+        lockfile = os.path.join(tmpdir, '.passwordstore.lock')
+
+        @contextmanager
+        def opt_lock(type):
+            if self.paramvals['lock'] == type:
+                unit_to_seconds = {"s": 1, "m": 60, "h": 3600}
+                timeout_secs = (int(self.paramvals['locktimeout'][:-1]) *
+                                unit_to_seconds[self.paramvals['locktimeout'][-1]])
```

Comment:
This is handled in parse_terms line 168 (all input validation happens ins that function):

‘’’
if not re.match('^[0-9]+[smh]$', self.paramvals['locktimeout']):
                 raise AnsibleError("{0} is not a correct value for locktimeout".format(self.paramvals['locktimeout']))
‘’’

Reply:
Then why not convert the value there? I think it would be a lot cleaner if there's only one piece of code that handles the parsing.

Reply:
> Then why not convert the value there? I think it would be a lot cleaner if there's only one piece of code that handles the parsing.

I disagree that it would be *a lot* cleaner. This way, we don’t introduce additional variables to be parsed to, and it’s consumed in exactly the place which knows how to interpret the values.

Please feel free to refactor the code later, I can’t spend much more time on this.